strapi/strapi
# chore: Support range selection in Content Manager ListView Table.

Closes https://github.com/strapi/strapi/issues/15799

### What does it do?
Support range selection/deselection on ListView when holding the keyboard shortcut `Shift + Click`

### Why is it needed?
Prevents the need to manually range select/deselect one item at a time for a desired action.

### How to test it?

1. Go to `Content-Type Builder`
2. Create a test component with a text field name and save.
3. Go to `Content Manager` and add 3 to 4 entries to the `test` component.
4. Select the first item by checking the checkbox.
5. Skip the second item to the third and try holding down the Shift key while checking the third item.
6. Notice that range selection works.
7. With multiple items selected, try deselecting at a desired position while holding the Shift key, notice that it deselects the range.


### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/15799

### Demo

https://user-images.githubusercontent.com/7220167/219575281-f2190ab9-1a20-40a2-8062-b78597eac365.mov


---
This code was written and reviewed by GitStart Community. Growing future engineers, one PR at a time.
